C/c4CAPI.cc

```diff
@@ -1059,6 +1060,14 @@ C4IndexUpdater* C4NULLABLE c4index_beginUpdate(C4Index* index, size_t limit, C4E
     return tryCatch<C4IndexUpdater*>(outError, [&] { return index->beginUpdate(limit).detach(); });
 }
 
+C4SliceResult c4index_getName(C4Index* index) noexcept {
+    return tryCatch<C4SliceResult>(nullptr, [&]{ return C4SliceResult(index->getName()); });
+}
+
+C4Collection* c4index_getCollection(C4Index* index) noexcept {
+    return tryCatch<C4Collection*>(nullptr, [&]{ return index->getCollection(); });
```

Comment:
c4Index::getCollection is a trivial getter that just returns a member, so there's no need to wrap a try/catch around this. (If you like, you could add `noexcept` to the C++ method declaration.)

Reply:
Oh okay I thought we might need it incase of null pointer access exception. But if not, I'll remove it

Reply:
Null pointer access is a signal/crash, not an exception, in C/C++.